packages/cli/templates/react/src/components/FieldPluginExample/index.spec.tsx

```diff
@@ -1,174 +1,15 @@
-    receive: (data) => {
-      if (isPluginLoadedMessage(data)) {
-        sendToFieldPlugin(
-          stateMessage({
-            action: 'loaded',
-            callbackId: data.callbackId,
-          }),
-        )
-      } else if (isValueChangeMessage(data)) {
-        model = data.model
-        sendToFieldPlugin(
-          stateMessage({
-            action: 'state-changed',
-            callbackId: data.callbackId,
-          }),
-        )
-      } else if (isModalChangeMessage(data)) {
-        isModalOpen = data.status
-        sendToFieldPlugin(
-          stateMessage({
-            action: 'state-changed',
-            callbackId: data.callbackId,
-          }),
-        )
-      } else if (isHeightChangeMessage(data)) {
-        height = data.height
-        sendToFieldPlugin(
-          stateMessage({
-            action: 'state-changed',
-            callbackId: data.callbackId,
-          }),
-        )
-      } else if (isAssetModalChangeMessage(data)) {
-        sendToFieldPlugin({
-          action: 'asset-selected',
-          uid,
-          filename: 'https://...',
-          field: '',
-          callbackId: data.callbackId,
-        })
-      } else if (isGetContextMessage(data)) {
-        sendToFieldPlugin({
-          action: 'get-context',
-          uid,
-          callbackId: data.callbackId,
-          story: {
-            content: {},
-          },
-        })
-      } else {
-        console.warn(
-          `Container received unknown message from plugin: ${JSON.stringify(
-            data,
-          )}`,
-        )
-      }
-    },
-  }
-}
-
-vi.mock('@storyblok/field-plugin', async () => {
-  const mod = await vi.importActual<typeof import('@storyblok/field-plugin')>(
-    '@storyblok/field-plugin',
-  )
-  const container = fakeContainer()
-  const mockedCreateFieldPlugin: CreateFieldPlugin = ({
-    onUpdateState,
-    validateContent,
-  }) =>
-    mod.internalCreateFieldPlugin({
-      onUpdateState,
-      validateContent,
-      postToContainer: (message) => {
-        return container.receive(message)
-      },
-      listenToContainer: (handleMessage) => {
-        container.setOnMessage((message) => {
-          return handleMessage(message)
-        })
-        return () => {}
-      },
-    })
-  return {
-    ...mod,
-    createFieldPlugin: mockedCreateFieldPlugin,
-  }
-})
-
-const setup = () => {
-  global.ResizeObserver = class ResizeObserver {
-    observe() {
-      // do nothing
-    }
-    unobserve() {
-      // do nothing
-    }
-    disconnect() {
-      // do nothing
-    }
-  }
-  vi.stubGlobal('parent', {
-    ...global.parent,
-  })
-  vi.stubGlobal('location', {
-    ...window.location,
-    search:
-      '?protocol=https%3A&host=plugin-sandbox.storyblok.com&uid=test-uid&preview=1',
-  })
-}
-
-describe('FieldPluginExammple', () => {
+describe('FieldPluginExample', () => {
   afterEach(() => {
     vi.unstubAllGlobals()
```

Comment:
After working closely on the test helper, I realized that we will need to provide additional documentation, maybe a comment that explains that vi.unstubAllGlobals() is needed if they use our helper `setup()` function. This at first glance might be rather unintuitive.

Reply:
Good point! Maybe we could do

```js
test('something', async () => {
  const { cleanUp } = setupFieldPlugin();
  
  // do stuff

  cleanUp();
});
```

And `cleanUp` is actually just `vi.unstubAllGlobals()` internally. What do you think?